app/command_test.go

```diff
@@ -221,7 +221,11 @@ func TestHandleCommandResponsePost(t *testing.T) {
 			Text: "<!here>",
 		},
 	}
+
+	// set and unset SkipSlackParsing here seems the nicest way as no separate response objects are created for every testcase.
+	resp.SkipSlackParsing = true
```

Comment:
Could you please elaborate why the change is needed? Why is `SkipSlackParsing` needed? The test should not modify the response.

Reply:
How I understand/guess the code works - im in no way a go professional - when an external MM command gets called, the json-response is unmarshalled and then `HandleCommandResponsePost` is called, with the model-objects filled from data from the unmarshalled json and `HandleCommandResponsePost` creates a post-object which then later is persisted into db.

Internal commands otoh create the response-object in their `DoCommand` function.

----

The tests test many other aspects of commands in the same function, reusing one `CommandResponse` for multiple `HandleCommandResponsePost` calls, modifying various fields of the response object...

+ https://github.com/mattermost/mattermost-server/blob/bb2c019cb8768dab4ec25e5fce3f6bcaa4e36bcb/app/command_test.go#L192
+ https://github.com/mattermost/mattermost-server/blob/bb2c019cb8768dab4ec25e5fce3f6bcaa4e36bcb/app/command_test.go#L207

The testcase for `/code` not getting Slack-Text-parsed was done by setting `command.trigger == "code"` and `HandleCommandResponsePost` having code that checked if the trigger was `code` to skip slack parsing.

Now with my changes the code no longer checks for `command.trigger == "code"`, but for ``response.SkipSlackParsing == true``.

Therefore I no longer need to modify the `command.trigger` to `"code"`, but need to set ``response.SkipSlackParsing = true``.

Normally this is set either by json unmarshalling or by `DoCommand` in command_code.go

Because I do not get a `/code` command's response instance in this test, I need to set ``response.SkipSlackParsing = true`` just before the `HandleCommandResponsePost` call.

To prevent unexpected side-effects on possible further testcases that may be appended to the function I also set ``response.SkipSlackParsing = false`` after the call to `HandleCommandResponsePost`, to restore default behavior.

tl;dr: it seems that all parameters are test-inputs except for `post`, which I do not modify.

Reply:
Got it. Thank you very much for the detailed explanation. I missed that we are using the same response for every test. (Which is a bad practice)